src/shared/hooks/useSearchQuery.ts

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
+import { useNexusContext } from '@bbp/react-nexus';
+import * as bodybuilder from 'bodybuilder';
+import { Resource } from '@bbp/nexus-sdk';
+
+import useAsyncCall, { AsyncCall } from './useAsynCall';
+import { parseURL } from '../utils/nexusParse';
+import { SearchResponse } from '../types/search';
+
+// TODO move to global default list
+const DEFAULT_PAGE_SIZE = 20;
+
+export type SerializedFacetMap = {
+  [propertyKey: string]: string[];
+};
+
+export const serializeSearchFacets = (
+  facetMap: UseSearchProps['facetMap']
+): SerializedFacetMap => {
+  return Array.from(facetMap || []).reduce((memo, [key, value]) => {
+    memo[value.propertyKey] = [...value.value.values()];
+    return memo;
+  }, {} as { [propertyKey: string]: any });
+};
+
+export const parseSerializedSearchFacets = (
+  facetMap: UseSearchProps['facetMap'],
+  serializedSearchFacets: SerializedFacetMap
+) => {
+  facetMap?.forEach(value => {
+    const serializedValue = serializedSearchFacets[value.propertyKey];
+    if (serializedValue) {
+      value.value.clear();
+      serializedValue.forEach(key => {
+        value.value.add(key);
+      });
+    }
+  });
+  return facetMap;
+};
+
+export enum SortDirection {
+  DESCENDING = 'desc',
+  ASCENDING = 'asc',
+}
+
+export type UseSearchProps = {
+  query?: string;
+  sort?: {
+    key: string;
+    direction: SortDirection;
+  };
+  pagination?: {
+    from: number;
+    size: number;
+  };
+  facetMap?: Map<
+    string,
+    {
+      propertyKey: string;
+      label: string;
+      type: 'terms';
+      value: Set<string>;
+    }
+  >;
+};
+
+export const DEFAULT_SEARCH_PROPS = {
+  pagination: { from: 0, size: DEFAULT_PAGE_SIZE },
+  sort: {
+    key: '_createdAt',
+    direction: SortDirection.DESCENDING,
+  },
+};
+
+export default function useSearchQuery(selfURL?: string | null) {
+  const [searchProps, setSearchProps] = React.useState<UseSearchProps>({
+    ...DEFAULT_SEARCH_PROPS,
+  });
+  const {
+    query,
+    sort = DEFAULT_SEARCH_PROPS.sort,
+    pagination = DEFAULT_SEARCH_PROPS.pagination,
+    facetMap = new Map<
+      string,
+      {
+        propertyKey: string;
+        label: string;
+        type: 'terms';
+        value: Set<string>;
+      }
+    >(),
+  } = searchProps;
+
+  const nexus = useNexusContext();
+
+  const searchNexus = async () => {
+    if (!selfURL) {
+      return null;
+    }
+
+    // TODO fix query to match @id's as well
+    const matchQuery = query
+      ? ['wildcard', '_original_source', `${query}*`]
+      : ['match_all', {}];
+
+    const body = bodybuilder();
+
+    body
+      // @ts-ignore
```

Comment:
Just checking you didn't forget to remove it :)

Reply:
we'll have to update ts to support what I'm trying to do here, I wrote a comment